scalene/scalene_preload.py

```diff
@@ -35,10 +34,20 @@ def get_preload_environ(args: argparse.Namespace) -> Dict[str, str]:
 
         elif sys.platform == "linux":
             if args.memory:
-                # Prepend the Scalene library to the LD_PRELOAD list, if any
-                new_ld_preload = os.path.join(
-                    scalene.__path__[0].replace(" ", r"\ "), "libscalene.so"
-                )
+
+                sanitized_path = scalene.__path__[0]
+                if escape_spaces:
+                    # This function is used in two places, in `setup_preload` (where escaping spaces causes problems)
+                    # and in `Scalene.__init__`. The latter creates a string by joining with spaces
+                    # to pass into `redirect_python`, so we do need spaces there.  
+                    sanitized_path = sanitized_path.replace(" ", r"\ ") 
+                    
+                    # NOTE: you can't use escape sequences inside an f-string pre-3.12 either
+                if 'LD_LIBRARY_PATH' in env:
```

Comment:
When would LD_LIBRARY_PATH ever get into env? I believe this is clobbering the LD_LIBRARY_PATH

I think we need to check if it is os.environ not env

(To verify the breakage pip install sentencepiece then try to import it while running scalene 1.5.45)

Reply:
Thanks for catching that! We actually also ran into it and fixed it [in a subsequent PR](https://github.com/plasma-umass/scalene/pull/867/files#diff-b7ca9f0b08184dee9528f9f80810d4c6037af641eec03d67affa0f656808d1d5R51). Incidentally, we had also been squashing LD_PRELOAD for a few years too.

Reply:
Awesome. Thanks!